pennylane/_device.py

```diff
@@ -312,6 +314,39 @@ def supported(self, name):
         """
         return name in self.operations.union(self.observables)
 
+    def supports_operation(self, operation):
+        """Checks if an operation is supported by this device.
+
+        Args:
+            operation (Operation/string): operation to be checked
```

Comment:
Better to use one of the following:

* `Operation,str` (I think this is what we use elsewhere in the library)

* `Union[Operation, str]` (preferred way to specify two types with type hinting in Python 3.5+)